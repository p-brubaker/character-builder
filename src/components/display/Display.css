.catchphrase-display {
  font-size: 2rem;
  font-family: cursive;
  text-align: center;
}
