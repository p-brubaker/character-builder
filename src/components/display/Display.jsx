import './Display.css'

function Display({ report, catchphrases }) {
  return (
    <div className="display">
      <p className="report">
        You have changed the head {report['Head']} times, the body {report['Middle']} times, and the
        pants {report['Bottom']} times. And nobody can forget your characters classic catchphrases:
      </p>
      <div className="catchphrases-display">
        {catchphrases.map((phrase) => (
          <p key={phrase}>{phrase}</p>
        ))}
      </div>
    </div>
  )
}

export default Display
