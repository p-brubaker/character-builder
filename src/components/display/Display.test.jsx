import { render } from '@testing-library/react'
import Display from '../display/Display'

it('should render the Display component', () => {
  const { container } = render(
    <Display
      report={{ Head: 0, Middle: 3, Bottom: 2 }}
      catchphrases={['Okey-Dokey', 'Well, I suppose...']}
    />
  )
  expect(container).toMatchSnapshot()
})
