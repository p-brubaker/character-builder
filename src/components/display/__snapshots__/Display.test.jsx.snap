// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Display component 1`] = `
<div>
  <div
    class="display"
  >
    <p
      class="report"
    >
      You have changed the head 
      0
       time
      s
      , the body
       3
       time
      s
      , and the pants
       2 
      time
      s
      . And nobody can forget your characters classic catchphrases:
    </p>
    <div
      class="catchphrases-display"
    >
      <p>
        Okey-Dokey
      </p>
      <p>
        Well, I suppose...
      </p>
    </div>
  </div>
</div>
`;
